scripts/download-vsix-from-circleci.js

```diff
@@ -0,0 +1,101 @@
+#!/usr/bin/env node
+
+const shell = require('shelljs');
+shell.set('-e');
+shell.set('+v');
+
+const { getVsixName, getLocalPathForDownload } = require('./publish-utils');
+const CIRCLECI_API_URI = 'https://circleci.com/api/v1.1';
+const circleciToken = process.env['CIRCLECI_TOKEN'];
+const circleciBuild = process.env['CIRCLECI_BUILD'];
+const nextVersion = process.env['SALESFORCEDX_VSCODE_VERSION'];
+
+// Check for CircleCI environment variables
+if (!circleciBuild) {
+  console.log(
+    'You must specify the CircleCI Build number (CIRCLECI_BUILD) that contains the artifacts that will be released.'
+  );
+  process.exit(-1);
+}
+
+if (!circleciToken) {
+  console.log(
+    'You must specify the CircleCI Token (CIRCLECI_TOKEN) that will allow us to get the artifacts that will be released.'
+  );
+  process.exit(-1);
+}
+
+// Check access to CircleCI & the project's builds
+const circleciMe = shell
+  .exec(`curl ${CIRCLECI_API_URI}/me?circle-token=${circleciToken}`, {
+    silent: true
+  })
+  .stdout.trim();
+
+const circleciMeJSON = JSON.parse(circleciMe);
+if (
+  circleciMeJSON.hasOwnProperty('message') &&
+  circleciMeJSON.message === 'You must log in first.'
+) {
+  console.log(
+    'Looks like your CircleCI Token is incorrect and does not grant you access'
+  );
+  process.exit(-1);
+}
+
+if (
+  !circleciMeJSON.hasOwnProperty('projects') ||
+  !circleciMeJSON.projects.hasOwnProperty(
+    'https://github.com/forcedotcom/salesforcedx-vscode'
+  )
+) {
+  console.log(
+    'Looks like your CircleCI Token does not grant you access to the salesforcedx-vscode builds'
+  );
+  process.exit(-1);
+}
+
+// Get artifact names from CircleCI build
+const cciArtifacts = shell
+  .exec(
+    `curl ${CIRCLECI_API_URI}/project/github/forcedotcom/salesforcedx-vscode/${circleciBuild}/artifacts?circle-token=${circleciToken}`,
+    {
+      silent: true
+    }
+  )
+  .stdout.trim();
+
+const buildArtifactsJSON = JSON.parse(cciArtifacts);
+if (buildArtifactsJSON && buildArtifactsJSON.length === 0) {
+  console.log(
+    `Looks like the CircleCI build number ${circleciBuild} did not generate any artifacts`
```

Comment:
I think we should include here that the build the user wants for getting the artifacts is 'build-all'.